### PR TITLE
Travis ci 29

### PR DESCRIPTION
**Pull request by andrewbird**
Next chunk of tests, mainly covering memory.

### CODE REVIEW

**Comment by ecm-pushbx** [comment]
Your commit message of adding my tests could list the links using the specific revision of which files you added. Instead of dpmitest/file/tip put https://hg.ulukai.org/ecm/dpmitest/file/42b4f7ec457d and also add https://hg.ulukai.org/ecm/lmacros/file/ede277bd4d21

**Comment by andrewbird** [comment]
Hi @ecm-pushbx, thanks for the feedback. I'll update the commit message.

File: test/test_dos.py [hunk 1]
```diff
@@ -6285,6 +6285,7 @@ def setUpClass(cls):
             "test_mfs_sfn_truename": KNOWNFAIL,
             "test_floppy_vfs": KNOWNFAIL,
             "test_pcmos_build": KNOWNFAIL,
+            "test_libi86_build": KNOWNFAIL,
```
Comment (stsp) [inline test/test_dos.py:6288]: What is the problem?

Comment (andrewbird) [reply test/test_dos.py:6288]: Some tests stalling and producing no output, see https://github.com/tkchia/libi86/issues/5. Those tests use dosemu's  `-input` option which seems to be a common factor, though I haven't started debugging it myself. It seems curious that the same tests work okay with FDPP and MS-DOS.

Comment (andrewbird) [reply test/test_dos.py]: That issue is now fixed in libi86

**Comment by andrewbird** [comment]
@stsp do you need any changes to this?

File: src/dosext/dpmi/dpmi.c [hunk 2]
```diff
@@ -2278,7 +2278,7 @@ static void do_int31(sigcontext_t *scp)
 	  _LWORD(edx) = 0;
 	  *buf = DPMI_VERSION;
 	  *(buf+1) = DPMI_DRIVER_VERSION;
-	  sprintf(buf+2, "DOSEMU Version %d.%d\n", VERSION_NUM, SUBLEVEL);
+	  snprintf(buf+2, 126, "DOSEMU Version %d.%d", VERSION_NUM, SUBLEVEL);
```
Comment (stsp) [inline src/dosext/dpmi/dpmi.c:2281]: 126??

Comment (andrewbird) [reply src/dosext/dpmi/dpmi.c:2281]: Seems to be the size of buffer remaining(128 - 2),  http://www.delorie.com/djgpp/doc/dpmi/api/310401.html

**Comment by stsp** [comment]
OK, thanks.
I thought this just awaits for
tkchia to fix his DPMI problems.

**Comment by andrewbird** [comment]
I wonder, are you okay with me creating Draft PRs, then upgrading them later when ready to merge? That's the process I used here, mainly because a Draft PR gets both MS-DOS and PP-DOS tests whereas a simple push to a branch only gets the PP-DOS tests run against it.

**Comment by stsp** [comment]
Why would I not be OK with that?
Drafts is the official github's functionality,
and I have to be OK with people using
any features that github proposes (or if
not - I should either disable them explicitly,
or change the site)

**Comment by andrewbird** [comment]
> Why would I not be OK with that?

No reason, I just wanted to check you were okay with it.

**Comment by stsp** [comment]
Have you noticed that tests failed?
https://travis-ci.com/github/dosemu2/dosemu2/builds/187506284

**Comment by andrewbird** [comment]
I had not, I'll check now, thanks!

**Comment by andrewbird** [comment]
I see that was the cron initiated test which is why I didn't get notified. It also means that the testing occurred on all four flavours of DOS. I see now that I need to mark the `test_libi86_build` test on FreeDOS 1.20 as a known failure, just like I do with the `test_pcmos_build`. This is due to its lack of the workaround removed in 43e66026 and its inability to redirect arbitrary directories that weren't already available as fatfs drives. I'll mark it as such in the next chunk of travis tests PR.

**Pull request merged**